src/resolve/connectionResolver.ts

```diff
@@ -108,10 +109,32 @@ export class ConnectionResolver {
     if (query.type === defaultRegistry.types.standardvalueset.name && members.length === 0) {
       const standardValueSetPromises = standardValueSet.fullnames.map(async (standardValueSetFullName) => {
         try {
-          const standardValueSetRecord: StdValueSetRecord = await this.connection.singleRecordQuery(
-            `SELECT Id, MasterLabel, Metadata FROM StandardValueSet WHERE MasterLabel = '${standardValueSetFullName}'`,
-            { tooling: true }
-          );
+          // The 'singleRecordQuery' method was having connection errors, using `retry` resolves this
+          // Note that this type of connection retry logic may someday be added to jsforce v2
+          // Once that happens this logic could be reverted
+          const standardValueSetRecord: StdValueSetRecord = await retry(async () => {
+            let result: StdValueSetRecord;
+
+            try {
+              result = await this.connection.singleRecordQuery(
+                `SELECT Id, MasterLabel, Metadata FROM StandardValueSet WHERE MasterLabel = '${standardValueSetFullName}'`,
+                { tooling: true }
+              );
+            } catch (err) {
+              // We exit the retry loop with `NotRetryableError` if we get an (expected) unsupported metadata type error
+              const error = err as Error;
+              if (error.message.includes('either inaccessible or not supported in Metadata API')) {
+                this.logger.debug('Expected error:', error.message);
+                throw new NotRetryableError(error.message);
+              }
+
+              // Otherwise throw the err so we can retry again
+              throw err;
+            }
+
+            return result;
```

Comment:
if you return the result from the try {} block, that would avoid the `let` , the assignment, and possibly the `await`